tests/unit/dom/trigger-key-event-test.js

```diff
@@ -123,4 +123,110 @@ module('DOM Helper: triggerKeyEvent', function(hooks) {
 
     assert.verifySteps(['keypress']);
   });
+
+  test('The value of the `event.key` is properly inferred from the given keycode and modifiers', async function(assert) {
+    element = buildInstrumentedElement('div');
+    async function checkKey(keyCode, key, modifiers) {
+      let handler = e => {
+        assert.equal(e.key, key);
+      };
+      element.addEventListener('keydown', handler);
+      await triggerKeyEvent(element, 'keydown', keyCode, modifiers);
+      element.removeEventListener('keydown', handler);
+    }
+    await checkKey(8, 'Backspace');
+    await checkKey(9, 'Tab');
+    await checkKey(13, 'Enter');
+    await checkKey(16, 'Shift');
+    await checkKey(17, 'Control');
+    await checkKey(18, 'Alt');
+    await checkKey(20, 'CapsLock');
+    await checkKey(27, 'Escape');
+    await checkKey(32, '');
+    await checkKey(37, 'ArrowLeft');
+    await checkKey(38, 'ArrowUp');
+    await checkKey(39, 'ArrowRight');
+    await checkKey(40, 'ArrowDown');
+    await checkKey(91, 'Meta');
+    await checkKey(93, 'Meta');
+    await checkKey(187, '=');
+    await checkKey(189, '-');
+    await checkKey(65, 'a');
+    await checkKey(90, 'z');
+    await checkKey(65, 'A', { shiftKey: true });
+    await checkKey(90, 'Z', { shiftKey: true });
+    assert.verifySteps([
```

Comment:
Btw, verifing steps like this is pretty annoying. Is there a way to avoid this?

Reply:
Don't build an "instrumented element", thats why you have to verify in this manner (because the instrumented elements also assert no other events are fired via steps).